### PR TITLE
Add reasoning fields.

### PR DESCRIPTION
**Pull request by eugenk**
~~Do not review just yet. I still need to figure out if there is more to come.~~

This is now ready for review. It is the ontohub-models-part of https://github.com/spechub/Hets/pull/1836.

### CODE REVIEW

File: app/models/conjecture.rb [hunk 1]
```diff
@@ -2,17 +2,16 @@
 
 # The Conjecture model
 class Conjecture < Sentence
-  EVALUATION_STATES = %w(not_yet_enqueued enqueued processing
-                         finished_successfully finished_unsuccessfully).freeze
-  REASONING_STATUSES ||= %w(OPN ERR UNK RSO THM CSA CSAS CONTR).freeze
+  PROOF_STATUSES ||= %w(OPN ERR UNK RSO THM CSA CSAS CONTR).freeze
```
Comment (phyrog) [inline app/models/conjecture.rb]: Why is this conditionally assigned?

File: app/models/proof_attempt.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 # The ProofAttempt model
 class ProofAttempt < ReasoningAttempt
+  PROOF_STATUSES ||= %w(OPN ERR UNK RSO THM CSA CSAS).freeze
```
Comment (phyrog) [inline app/models/proof_attempt.rb]: Same here

File: db/migrate/20160902111740_initial_schema.rb [hunk 3]
```diff
@@ -536,9 +592,9 @@ def create_trigger_to_delete_parent(child_table, parent_table)
       column :text, String, null: false
     end
 
-    # REASONING_STATUSES ||= %w(OPN ERR UNK RSO THM CSA CSAS).freeze
-    # create_enum :reasoning_status_on_conjecture_type,
-    #   [*REASONING_STATUSES, 'CONTR']
+    # PROOF_STATUSES ||= %w(OPN ERR UNK RSO THM CSA CSAS).freeze
```
Comment (phyrog) [inline db/migrate/20160902111740_initial_schema.rb]: `CONTR` is element of `Conjecture::PROOF_STATUSES`, why the split here?

Comment (eugenk) [reply db/migrate/20160902111740_initial_schema.rb]: CONTR means that there are proof attempts that say the conjecture is definitely valid and at the same time there are proof attempts that say the conjecture is definitely invalid. So there are contradictory proof attempts. This status does not make sense on a single ProofAttempt (and indicates a bug in the reasoning system).

File: app/models/conjecture.rb (hunk 1 above)
Comment (eugenk) [reply app/models/conjecture.rb]: I'm not sure any more. I think this file is required many times and from the second require statement on, the system complains about a redefined constant.

**Comment by eugenk** [comment]
Rebased on master

**Pull request merged**